app/controllers/carts_controller.rb

```diff
@@ -11,20 +14,29 @@ def show
   end
 
   # GET /carts/new
-  def new
-    @cart = Cart.new
+   def new
```

Comment:
I think you should add that user should be logged in before creating cart.

Reply:
authenticate_user! used in application controller